api_v1/serializers.py

```diff
@@ -0,0 +1,49 @@
+from rest_framework import serializers
+
+from ports.models import Port, Maintainer, Variant, BuildHistory, Builder
+
+
+class MaintainerSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Maintainer
+        fields = ('name', 'domain', 'github')
+
+
+class VariantSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Variant
+        fields = ('variant', )
+
+
+class PortSerializer(serializers.ModelSerializer):
+    maintainers = MaintainerSerializer(read_only=True, many=True)
+    variants = serializers.SerializerMethodField()
+
+    class Meta:
+        model = Port
+        fields = ('name', 'portdir', 'categories', 'maintainers', 'version', 'variants')
+
+    def get_variants(self, obj):
+        qs = Variant.objects.filter(port=obj)
+        variants = VariantSerializer(qs, many=True, read_only=True, context=self.context).data
+        return variants
+
+
+class BuilderSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Builder
+        fields = ('name', )
+
+
+class BuildHistorySerializer(serializers.ModelSerializer):
+    builder_name = BuilderSerializer(read_only=True)
+
+    class Meta:
+        model = BuildHistory
+        fields = ('builder_name', 'build_id', 'status', 'time_start', 'time_elapsed', 'watcher_id')
+
+
+class PortListSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Port
+        fields = ('name', )
```

Comment:
This is something with django rest framework. For list/ tuple with just one item- it forces me to use a trailing comma, otherwise throws an error. I tried using using a single item without a tuple, that does not work either.

Reply:
Sorry, python docs confirm this.

Reply:
Sorry, python docs confirm this.

Reply:
Sorry, python docs confirm this.

Reply:
@arjunsalyan You cannot have a tuple as `x = ('a')` but both `x = 'a',` or `x = ('a',)` are valid, it's not an issue of Django but python forbids it. In case of more than one element, you can have `y = ('a', 'b')` as well as `y = ('a', 'b',)`. In the case of one element, it cannot distinguish this syntax from simply having an extra bracket in arithmetics, e.g., `a = (1)` vs `a = 1`.

A helpful link [here](https://stackoverflow.com/questions/7992559/what-is-the-syntax-rule-for-having-trailing-commas-in-tuple-definitions) and you'll find more information from further googling.